src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -103,7 +98,12 @@ private static boolean samRecordOverlaps(final SAMRecord record, final List<Simp
             return true;
         }
         for (SimpleInterval interval : intervals) {
-            if (interval.overlaps(record)) {
+            if (record.getReadUnmappedFlag() && record.getAlignmentStart() != SAMRecord.NO_ALIGNMENT_START) {
```

Comment:
I think it's OK as he merged the equivalent code into dataflow a couple of days ago: https://github.com/broadinstitute/hellbender/pull/880.